bioio/bio_image.py

```diff
@@ -1174,3 +1227,63 @@ def imread(
         np.ndarray.
     """
     return _construct_img(image, scene_id, **kwargs).data
+
+
+from typing import Any, Dict
+
+
+def _check_plugin_support(
+    plugin: PluginEntry, image: biob.types.ImageLike, fs_kwargs: Dict[str, Any] = {}
+) -> Dict[str, Union[bool, Any]]:
+    """Helper function to check if a plugin supports the given image."""
+    try:
+        ReaderClass = plugin.metadata.get_reader()
+        return {
+            "supported": ReaderClass.is_supported_image(
+                image=image, fs_kwargs=fs_kwargs
+            ),
+            "error": None,
+        }
+    except Exception as e:
+        return {"supported": False, "error": str(e)}
+
+
+def plugin_feasibility_report(
+    image: biob.types.ImageLike,
+    fs_kwargs: Dict[str, Any] = {},
+    use_plugin_cache: bool = False,
+    **kwargs: Any,
+) -> Dict[str, Dict[str, Union[bool, Any]]]:
+    """
+    Generate a feasibility report for each plugin,
+    determining if it can handle the specified image.
+    """
+    plugins_by_ext = get_plugins(use_cache=use_plugin_cache)
+    feasibility_report = {}
+
+    # Check each plugin for support
+    for plugins in plugins_by_ext.values():
+        for plugin in plugins:
+            plugin_name = plugin.entrypoint.name
+            feasibility_report[plugin_name] = _check_plugin_support(
+                plugin, image, fs_kwargs
+            )
+
+    # Additional check for ArrayLike support
+    try:
+        feasibility_report["ArrayLike"] = {
+            "supported": isinstance(image, get_args(MetaArrayLike) + (list,)),
```

Comment:
Is there not a function in bioio for checking ArrayLike support already? Or, is this code duplicated by some logic elsewhere that does `isinstance(image, get_args(MetaArrayLike) + (list,))`?

Reply:
resolved in [0e2c7e7](https://github.com/bioio-devs/bioio/pull/63/commits/0e2c7e751ee25009c648ef79eb29635999686540)